clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/PartitionConsumptionState.java

```diff
@@ -773,16 +787,21 @@ public long getLatestProcessedUpstreamRTOffset(String kafkaUrl) {
     return latestProcessedUpstreamRTOffset;
   }
 
+  public long getLeaderOffset(String kafkaURL, PubSubTopicRepository pubSubTopicRepository) {
+    return getLeaderOffset(kafkaURL, pubSubTopicRepository, false);
+  }
+
   /**
    * The caller of this API should be interested in which offset currently leader should consume from now.
    * 1. If currently leader should consume from real-time topic, return upstream RT offset;
+   *    If Global RT DIV is enabled, use the value of LCRO from the Global RT DIV from StorageEngine.
    * 2. if currently leader should consume from version topic, return either remote VT offset or local VT offset, depending
    *    on whether the remote consumption flag is on.
    */
-  public long getLeaderOffset(String kafkaURL, PubSubTopicRepository pubSubTopicRepository) {
+  public long getLeaderOffset(String kafkaURL, PubSubTopicRepository pubSubTopicRepository, boolean useLcro) {
     PubSubTopic leaderTopic = getOffsetRecord().getLeaderTopic(pubSubTopicRepository);
     if (leaderTopic != null && !leaderTopic.isVersionTopic()) {
-      return getLatestProcessedUpstreamRTOffset(kafkaURL);
+      return (useLcro) ? getLatestConsumedRtOffset(kafkaURL) : getLatestProcessedUpstreamRTOffset(kafkaURL);
     } else {
       return consumeRemotely()
           ? getLatestProcessedUpstreamVersionTopicOffset()
```

Comment:
In this case should we check useLcro and get the latest consumed vt offset?
Also, I would suggest useLcro -> useLatestConsumedVtOffset to be more clear

Reply:
Can you add some java doc ofr latestConsumedXXOffset vs latestProcessedXXOffset? Sometimes it is getting people consumed and will take sometime to figure out which one to use and which one takes priority.